### PR TITLE
utility method to extract a pagination cursor from a jsonapi document

### PR DESCRIPTION
**Pull request by 21echoes**
For paginated jsonapi documents, there will often be pagination links present, typically at `links.next` in the document. Provide a utility method to easily pull out that link's cursor query param.

### CODE REVIEW

**Comment by 21echoes** [comment]
implements the desired changes of https://github.com/Patreon/patreon-python/pull/1 as a new explicit method, rather than an optional parameter passed to another method

File: patreon/api.py [hunk 1]
```diff
@@ -48,6 +48,37 @@ def fetch_page_of_pledges(self, campaign_id, page_size,
             fields=fields
         ))
 
+    @staticmethod
+    def extract_cursor(jsonapi_document, cursor_path='links.next'):
+        def head_and_tail(path):
```
Comment (LiraNuna) [inline patreon/api.py]: This looks better implemented with an iterator

File: patreon/api.py [hunk 2]
```diff
@@ -48,6 +48,37 @@ def fetch_page_of_pledges(self, campaign_id, page_size,
...
+        if current_dict is None or (head is not None and tail is None):
+            return None
+        # Path stopped before leaf was reached
+        elif current_dict and type(current_dict) != str:
+            raise Exception('Provided cursor path did not result in a link', current_dict)
+
+        link = current_dict
+        query_string = urlparse(link).query
+        parsed_query_string = parse_qs(query_string)
+        if 'page[cursor]' in parsed_query_string:
```
Comment (LiraNuna) [inline patreon/api.py]: nit: I usually go for exceptional-case -> truth:
```python
if 'page[cursor]' not in parsed_query_string:
    return None

return return parsed_query_string['page[cursor]'][0]
```

**Comment by LiraNuna** [comment]
Looks great - As this utility matures I think a `PagedResponse` sort of interface would be better but I understand this is a temporary solution.

**Pull request merged**